### ISSUE TITLE
upgrade accuracy

### ISSUE THREAD

**Issue by ISSE0116** [ISSE0116/ImageNet-1k_finetuned_311classes#2]
**現時点での最大Accは0.920**

## 前提(fine-tuning)
- 311クラスでImageNetで学習済みのネットワークをfinetuningした
  - 各クラス50枚で学習(train : 12440, val : 3110)
    - resnet18 :       Acc  0.488    [batch:450, epoch:100, lr:0.001]
    - resnet50 :       Acc  0.591    [batch:64, epoch:25, lr:0.001]
    - resnet101 :     Acc  0.624    [batch:64, epoch:25, lr:0.001]
    - densenet121 : Acc 0.565     [batch:64, epoch:25, lr:0.001]
    - densenet161 : Acc 0.608     [batch:32, epoch:25, lr:0.001]
    - densenet201 : Acc 0.578     [batch:32, epoch:100, lr:0.001]
  - 各クラス1300枚で学習(train : 317123, val : 79281)
    - resnet18 :       Acc  0.631    [batch:450, epoch:25, lr:0.001]
    - resnet50 :       Acc  0.713    [batch:64, epoch:25, lr:0.001]
    - resnet101 :     Acc  0.734     [batch:64, epoch:25, lr:0.001]
    - densenet121 : Acc 0.695     [batch:64, epoch:25, lr:0.001]
    - densenet161 : --
    - densenet201 : --

## 試したこと

**1. 最適化手法Adamを使う**
- SGD から Adamに変更
- 微かに精度が向上した
  - 各クラス50枚で学習
    - resnet50 : Acc 0. 590 → 0.613
- 以降、最適化手法はAdamを使用した

**2. 転移学習(出力層のみ)で試す**
- finetuningに比べ制度はやや向上
- 学習コストも低く、使用メモリも抑えられる為こっちの方が使い勝手が良いかも
  - 各クラス50枚で学習
    - resnet18 : Acc  0.542 [batch:512, epoch:100, lr:0.01]
    - resnet50 : Acc  0.613 [batch:512, epoch:50, lr:0.001]
    - resnet101 : --
    - densenet121 : --
    - densenet161 : Acc 0.641 [batch:256, epoch:50, lr:0.001]
    - densenet201 : --
    - efficientnetb7 : Acc 0.637 [batch:256, epoch:50, lr:0.001]
  - 各クラス1300枚で学習
    - resnet18 : Acc 0.575        [batch:512, epoch:10, lr:0.001]
    - resnet50 : Acc 0.752        [batch:512, epoch:25, lr:0.001]
    - resnet101 : --
    - densenet121 : --
    - densenet161 : Acc 0.767   [batch:256, epoch:10, lr:0.001]
    - densenet201 : --
    - efficientnetb7 : --

**3. top5で試す(出力層のみ)**
- 当たり前だが精度は大幅に向上
  - 各クラス50枚で学習
    - resnet18 : Acc 0.73           [batch:1024, epoch:10, lr:0.001]
    - resnet50 : Acc 0.83           [batch:512, epoch:25, lr:0.001]
    - resnet101 : --
    - densenet121 : --
    - densenet161 : --
    - densenet201 : --
    - efficientnetb7 : --
  - 各クラス1300枚で学習
    - resnet18 : 
    - resnet50 : Acc 0.908         [batch:512, epoch:10, lr:0.001]
    - resnet101 : --
    - densenet121 : --
    - densenet161 : Acc 0.920   [batch:256, epoch:10, lr:0.001]
    - densenet201 : --
    - efficientnetb7 : --

**Comment by ISSE0116** [comment]
## 50class ver 試してみた
**1. 転移学習(出力層のみ)で試す**
- resnet50 : Acc  0.8211 [batch:512, epoch:50, lr:0.001]
